### ISSUE TITLE
Keras 2.2.2 > Training Loss increases for large data sets (surprisingly mse keeps low).

### ISSUE THREAD

**Issue by krishnadasar-sudheer-kumar** [keras-team/keras#10932]
The training loss increases for large datasets for a regression model. There are only 4 features in the input set of 22M records. The model's training loss increases after 12M rows (jumps from low loss of 30 to 300 and finally settles around 190). I experimented with regularizers and learning rate but doesn't help much. lr of both 1e-3 and 1e-4 didn't have much effect. The lr of 1e-5 doesn't decrease loss but decreases the accuracy.
 
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
model.add(layers.Dense(4, kernel_regularizer = regularizers.l1_l2(l1=0.001, l2=0.001), activation = 'relu', input_shape = (input_shape,)))
  model.add(layers.Dense(1))
  model.compile(optimizer = 'rmsprop', loss = 'mse', metrics = ['mae'])
# training_set = 22M rows.
test_history = model.fit(training_set, training_label, epochs = 30, batch_size = 1000)
